test_util/launch.py

```diff
@@ -1,20 +1,282 @@
+            'delete_with_stack': generate_key,
+            'private_key': private_key,
+            'key_name': key_name,
+            'user': config.get('ssh_info', dict()).get('user', None)}
+
+    def test(self, info, test_cmd):
+        ssh_info = info['ssh']
+        for param in ['user', 'private_key']:
+            if param not in ssh_info:
+                raise LauncherError('MissingInfo', 'Cluster was created without providing ssh_info: {}'.format(param))
+        details = self.describe(info)
+        test_host = details['masters'][0]['public_ip']
+        # TODO: section should be generalized for use by other pr
+        with ssh.tunnel.tunnel(ssh_info['user'], ssh_info['private_key'], test_host) as test_tunnel:
+            return test_util.runner.integration_test(
+                tunnel=test_tunnel,
+                dcos_dns=test_host,
+                master_list=[m['private_ip'] for m in details['masters']],
+                agent_list=[a['private_ip'] for a in details['private_agents']],
+                public_agent_list=[a['private_ip'] for a in details['public_agents']],
+                aws_access_key_id=info['provider']['access_key_id'],
+                aws_secret_access_key=info['provider']['secret_access_key'],
+                region=info['provider']['region'],
+                test_cmd=test_cmd)
+
+    def get_stack(self, info):
+        """Returns the correct class interface depending how the AWS CF is configured
+        NOTE: only supports Simple Cloudformation currently
+        """
+        try:
+            return DcosCfSimple(info['stack_name'], self.boto_wrapper)
+        except Exception as ex:
+            raise LauncherError('StackNotFound', '{} is not accessible'.format(info['stack_name'])) from ex
+
+
+def get_launcher(launcher_type, provider_info):
+    """Returns a launcher given a launcher type (string) and a dictionary of
+    appropriate provier_info
+    """
+    if launcher_type == 'cloudformation':
+        check_keys(provider_info, ['region', 'access_key_id', 'secret_access_key'])
+        boto_wrapper = BotoWrapper(
+            provider_info['region'], provider_info['access_key_id'], provider_info['secret_access_key'])
+        return AwsCloudformationLauncher(boto_wrapper)
+    else:
+        raise LauncherError('UnsupportedAction', 'Launcher type not supported: {}'.format(launcher_type))
+
+
+def convert_host_list(host_list):
+    """ Makes Host tuples more readable when using describe
+    """
+    return [{'private_ip': h.private_ip, 'public_ip': h.public_ip} for h in host_list]
+
+
+def check_keys(user_dict, key_list):
+    missing = [k for k in key_list if k not in user_dict]
+    if len(missing) > 0:
+        raise LauncherError('MissingInput', 'The following keys were required but'
+                            'not provided: {}'.format(repr(missing)))
+
+
+def main(argv):
+    args = docopt(__doc__, argv=argv, version=DCOS_LAUNCH_VERSION)
+
+    try:
+        if args['create']:
```

Comment:
nit: Rather than have this all be "indented"  by the try (getting to high levels of indent here) might make sense to move the dispatch to a second helper function this calls.